packages/fx-core/tests/plugins/resource/simpleauth/helper.ts

```diff
@@ -135,8 +147,52 @@ export class TestHelper {
           short: "hello-app",
         },
       },
+      projectSettings: {
+        solutionSettings: {
+          activeResourcePlugins: [
+            Constants.AadAppPlugin.id,
+            Constants.FrontendPlugin.id,
+            Constants.SimpleAuthPlugin.id,
+          ],
+        },
+      },
     } as unknown as PluginContext;
 
     return pluginContext;
   }
+
+  static mockSolutionUpdateArmTemplates(
+    mockedData: any,
+    template: ScaffoldArmTemplateResult
+  ): ScaffoldArmTemplateResult {
+    return {
+      Modules: template.Modules,
+      Orchestration: {
+        ParameterTemplate: {
+          Content: TestHelper.generateBicepFiles(
+            template.Orchestration.ParameterTemplate!.Content,
+            mockedData
+          ),
+        },
+        ModuleTemplate: {
+          Content: TestHelper.generateBicepFiles(
+            template.Orchestration.ModuleTemplate.Content,
+            mockedData
+          ),
+          Outputs: template.Orchestration.ModuleTemplate.Outputs,
+        },
+        OutputTemplate: {
+          Content: TestHelper.generateBicepFiles(
+            template.Orchestration.OutputTemplate!.Content,
+            mockedData
+          ),
+        },
+      },
+    } as ScaffoldArmTemplateResult;
+  }
+
+  static generateBicepFiles(templateString: string, context: any): string {
```

Comment:
can we use the generate bicep file logic in resource plugin? The function should be tested by other unit tests, so we can be confident to use this function.

Reply:
I think this method can be put in a common utility **for test only** where all resource plugins can leverage it for testing purpose. I don't think we should use the `generateBicepFiles()` method in `src/common/tools.ts` since not only do they have different signature but they serve for different usage scenario.

Reply:
The major concern is, when we add extra handlebars helper in the future, we also need to add that helper here. Is there any scenario that the implementation here will be different with the one in src/common/tools.ts?
If you're concerned that we may not be aware if the behavior of the one in src/common/tools.ts changes, the function should also have unit tests to cover this as mentioned previously.

Reply:
Fixed.